frontend/src/components/mine/MineDashboard.js

```diff
@@ -111,10 +111,19 @@ export class MineDashboard extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const { activeTab } = nextProps.match.params;
+    const { id, activeTab } = nextProps.match.params;
     if (activeTab !== this.props.activeTab) {
       this.setState({ activeTab });
     }
+    if (this.props.match.params.id !== nextProps.match.params.id) {
```

Comment:
suggestion to DRY up the code, move the logic inside this block into its own function and call it in both `componentDidMount` and `componentWillReceiveProps`

Very happy with this update, by the way, good find.

Reply:
fixed